File: StoryFlow_Shared/Models/GetStoryDto.cs
using StoryFlow_Shared.Enums;

namespace StoryFlow_Shared.Models
{
    public class GetStoryDto
    {
        public int Id { get; set; }
        public string? PolishStory { get; set; }
        public string? EnglishStory { get; set; }
        public string? PolishTitle { get; set; }
        public string? EnglishTitle { get; set; }
        public string? PolishDescription { get; set; }
        public string? EnglishDescription { get; set; }
        public int MaxPoints { get; set; }
        public StoryCategory? StoryCategory { get; set; }
        public StorySize? StorySize { get; set; }
        public LanguageLevel? LanguageLevel { get; set; }
        public ICollection<GetSentenceDto> Sentences { get; set; } = new List<GetSentenceDto>();
        public ICollection<GetUserStoryDto> UserStories { get; set; } = new List<GetUserStoryDto>();
        public GetQuizDto? Quiz { get; set; }
    }
}

File: StoryFlow_Client/src/app/models/get-story-dto.ts
import { LanguageLevel } from "../enums/language-level";
import { StoryCategory } from "../enums/story-category";
import { StorySize } from "../enums/story-size";
import { GetQuizDto } from "./get-quiz-dto";
import { GetSentenceDto } from "./get-sentence-dto";
import { GetUserStoryDto } from "./get-user-story-dto";
import { GetStoryPoint } from "./get-story-points";


export interface GetStoryDto {
    id: number;
    polishStory: string | null;
    englishStory: string | null;
    polishTitle: string | null;
    englishTitle: string | null;
    polishDescription: string | null;
    englishDescription: string | null;
    storyCategory: StoryCategory | null;
    storySize: StorySize | null;
    storyPointId: number;
    storyPoint: GetStoryPoint | null;
    languageLevel: LanguageLevel | null;
    sentences: GetSentenceDto[];
    userStories: GetUserStoryDto[];
    quiz: GetQuizDto;
    numberOfQuestions : number;
}